frontend/lib/framework/Context.cpp

```diff
@@ -639,32 +639,39 @@ void Context::setFilePathForModuleId(ID moduleID, UniqueString path) {
     printf("%i SETTING FILE PATH FOR MODULE %s -> %s\n", queryTraceDepth,
            moduleIdSymbolPath.c_str(), path.c_str());
   }
+  // check that querying the module ID works...
+  UniqueString gotPath;
+  bool ok = filePathForId(moduleID, gotPath);
   #ifndef NDEBUG
-    // check that querying the module ID works...
-    UniqueString gotPath;
-    bool ok = filePathForId(moduleID, gotPath);
     CHPL_ASSERT(ok);
+  #endif
 
-    // ... and gives the same path
+  // ... and gives the same path
 
-    // Note: if this check causes problems in the future, it could
-    // be removed, or we could wire up setFileText used in tests
-    // to work with the LLVM VirtualFilesystem
+  // Note: if this check causes problems in the future, it could
+  // be removed, or we could wire up setFileText used in tests
+  // to work with the LLVM VirtualFilesystem
 #if LLVM_VERSION_MAJOR <= 11
     llvm::SmallVector<char, 64> realPath, realGotPath;
 #else
     llvm::SmallVector<char> realPath, realGotPath;
 #endif
-    std::error_code errPath;
-    std::error_code errGotPath;
-    errPath = llvm::sys::fs::real_path(path.str(), realPath);
-    errGotPath = llvm::sys::fs::real_path(gotPath.str(), realGotPath);
-    if (errPath || errGotPath) {
-      // ignore the check if there were errors
-    } else {
-      CHPL_ASSERT(realPath == realGotPath);
-    }
-  #endif
+  std::error_code errPath;
+  std::error_code errGotPath;
+  errPath = llvm::sys::fs::real_path(path.str(), realPath);
+  errGotPath = llvm::sys::fs::real_path(gotPath.str(), realGotPath);
+  if (!ok || errPath || errGotPath) {
+    // ignore the check if there were errors
+  } else
+    // Check for duplicate modules names, but skip over bundled modules
+    // since we don't necessarily want to preclude the user from using
+    // the same names that we happen to have chosen.
+    if (realPath != realGotPath &&
+        !parsing::idIsInBundledModule(this, moduleID)) {
+      error(moduleID,
+            "Redefinition of module '%s' (the original was defined in '%s')",
+            moduleIdSymbolPath.c_str(), path.c_str());
+  }
```

Comment:
Perhaps you want to check if either the old or the new path is in a bundled module? You can also achieve this by calling `filePathIsInBundledModule` with `realPath` and `realGotPath` (or if that doesn't work, on `path` and `gotPath`?)

Reply:
I wondered about this as well and didn't do it primarily because I didn't know about that routine and didn't have an ID for the other module (or know how to get one); secondarily because just this single check made the existing tests that failed without it pass.

I admittedly treated this exercise very much as a black box problem:  "This assertion is triggering, but should be generating a user-facing error message instead.  What do I need to do to make that happen and keep tests passing?"

If I were to add an additional check, I'd want to add a new test that exercised that check, but don't have any intuition what that program would need to be—do you?

Reply:
> If I were to add an additional check, I'd want to add a new test that exercised that check, but don't have any intuition what that program would need to be—do you?

I guess you'd need to find a way to get a non-bundled module to parse _before_ a bundled module (if such a thing were even possible). If it's possible, I would imagine a require statement might do the trick?

Reply:
It seems like if a non-bundled module was parsed before a bundled module, we'd definitely want the warning, though, wouldn't we?  I.e., nothing in Chapel's internal modules depended on the bundled module; yet something in the user's code is (presumably incorrectly) relying on both the bundled module and the user module?

The `require` idea was a good suggestion.  Here's a case that generates the redefinition error for me (and makes me believe that having the error is better than not for this case):


testit.chpl:
```chapel
module Version {
  use M;
  writeln("In My version");
}
```

M.chpl:
```chapel
require "/Users/bradc/chapel/modules/standard/Version.chpl";

writeln("In M");
```

```bash
$ chpl testit.chpl
testit.chpl:1: error: Redefinition of module 'Version' (the original was defined in '/Users/bradc/chapel/modules/standard/Version.chpl')
```

Reply:
> It seems like if a non-bundled module was parsed before a bundled module, we'd definitely want the warning, though, wouldn't we? 

That's a good point, I agree with you.